2010-05-02  Ulrich Drepper  <drepper@redhat.com>

	* misc/mntent_r.c (encode_name): The slow loop handles newlines so we
	should recognize them as an abort condition.
	Patch by Jan Lieskovsky <jlieskov@redhat.com>.

diff --git a/misc/mntent_r.c b/misc/mntent_r.c
index 829750b..9598528 100644
--- a/misc/mntent_r.c
+++ b/misc/mntent_r.c
@@ -194,7 +194,7 @@ weak_alias (__getmntent_r, getmntent_r)
     const char *rp = name;						      \
 									      \
     while (*rp != '\0')							      \
-      if (*rp == ' ' || *rp == '\t' || *rp == '\\')			      \
+      if (*rp == ' ' || *rp == '\t' || *rp == '\n' || *rp == '\\')	      \
 	break;								      \
       else								      \
 	++rp;								      \
